src/main/java/org/opensearch/neuralsearch/ml/MLCommonsClientAccessor.java

```diff
@@ -874,47 +890,36 @@ private Map<String, String> extractConversationalAgentResult(MLOutput mlOutput,
                         continue;
                     }
 
-                    // Extract response containing dsl_query and agent_steps_summary
+                    // Extract response containing dsl_query or agent steps
                     if (!"response".equals(tensorName)) {
                         continue;
                     }
 
-                    Map<String, Object> dataMap = (Map<String, Object>) tensor.getDataAsMap();
-                    if (dataMap == null || !dataMap.containsKey("response")) {
+                    String responseJsonString = tensor.getResult();
+                    if (responseJsonString == null || responseJsonString.isBlank()) {
                         continue;
                     }
 
-                    Object responseObj = dataMap.get("response");
-                    if (!(responseObj instanceof String)) {
-                        continue;
-                    }
-
-                    String responseJsonString = (String) responseObj;
-
-                    if (responseJsonString.isBlank()) {
-                        continue;
-                    }
-
-                    // Extract JSON object from the response string (handles cases with text + JSON)
-                    // This will throw IllegalArgumentException if JSON is invalid
-                    Map<String, Object> responseMap = extractJsonObjectFromString(responseJsonString);
-
-                    // Extract dsl_query and agent_steps_summary from the response map
-                    Object dslQueryObj = responseMap.get(DSL_QUERY_FIELD_NAME);
-                    Object stepsSummaryObj = responseMap.get(AGENT_STEPS_FIELD_NAME);
+                    // Try to extract JSON object from the response string
+                    try {
+                        Map<String, Object> responseMap = extractJsonObjectFromString(responseJsonString);
```

Comment:
responseJsonString Can you rename this variable name to something meaning ful? Like which responeJson it is?